src/SystemCommands-VariableCommands/SycGenerateVariableAccessorCommand.class.st

```diff
@@ -22,13 +22,14 @@ SycGenerateVariableAccessorCommand >> defaultMenuItemName [
 
 { #category : 'execution' }
 SycGenerateVariableAccessorCommand >> execute [
-	
-	(ReGenerateAccessorsDriver new 
-		targetClass: variables first definingClass;
-		scopes: toolContext refactoringScopes;
-		selectedVariables: (variables collect: [:each | each name ]) ) runRefactoring
-	
 
+	(ReGenerateAccessorsDriver new
+		 targetClass: variables first definingClass;
```

Comment:
The #definingClass actually gives you the correct class side object for class side variables.
You can detect the class side by:
```Smalltalk
variables first definingClass isInstanceSide
```
No need to analyze the toolContext at all. Variables are first class objects with the full information.

Reply:
The goal is to make it easy to use for scripting.
When you call it in a playground it's not nice to have to get variable objects.
But maybe we can add some API to configure the driver with real variables and then the driver gets the name and side of the variable for the refactoring?